src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs

```diff
@@ -2,22 +2,160 @@
+            Assert.NotNull(control.FlatAppearance);
+            Assert.Same(control.FlatAppearance, control.FlatAppearance);
+            Assert.Equal(FlatStyle.Standard, control.FlatStyle);
+            Assert.Equal(Control.DefaultFont, control.Font);
+            Assert.Equal(control.Font.Height, control.FontHeight);
+            Assert.Equal(Control.DefaultForeColor, control.ForeColor);
+            Assert.False(control.HasChildren);
+            Assert.Equal(23, control.Height);
+            Assert.Null(control.Image);
+            Assert.Equal(ContentAlignment.MiddleCenter, control.ImageAlign);
+            Assert.Equal(-1, control.ImageIndex);
+            Assert.Empty(control.ImageKey);
+            Assert.Null(control.ImageList);
+            Assert.Equal(ImeMode.Disable, control.ImeMode);
+            Assert.Equal(ImeMode.Disable, control.ImeModeBase);
+            Assert.False(control.IsDefault);
+            Assert.NotNull(control.LayoutEngine);
+            Assert.Same(control.LayoutEngine, control.LayoutEngine);
+            Assert.Equal(0, control.Left);
+            Assert.Equal(Point.Empty, control.Location);
+            Assert.Equal(new Padding(3), control.Margin);
+            Assert.Equal(Size.Empty, control.MaximumSize);
+            Assert.Equal(Size.Empty, control.MinimumSize);
+            Assert.Equal(Padding.Empty, control.Padding);
+            Assert.Null(control.Parent);
+            Assert.Equal("Microsoft\u00AE .NET", control.ProductName);
+            Assert.Equal(new Size(75, 23), control.PreferredSize);
+            Assert.False(control.RecreatingHandle);
+            Assert.Null(control.Region);
+            Assert.True(control.ResizeRedraw);
+            Assert.Equal(75, control.Right);
+            Assert.Equal(RightToLeft.No, control.RightToLeft);
+            Assert.Null(control.Site);
+            Assert.Equal(new Size(75, 23), control.Size);
+            Assert.Equal(0, control.TabIndex);
+            Assert.True(control.TabStop);
+            Assert.Empty(control.Text);
+            Assert.Equal(ContentAlignment.MiddleCenter, control.TextAlign);
+            Assert.Equal(TextImageRelation.Overlay, control.TextImageRelation);
+            Assert.Equal(0, control.Top);
+            Assert.Null(control.TopLevelControl);
+            Assert.True(control.UseMnemonic);
+            Assert.True(control.UseCompatibleTextRendering);
+            Assert.True(control.UseVisualStyleBackColor);
+            Assert.True(control.Visible);
+            Assert.Equal(75, control.Width);
+
+            Assert.False(control.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        public void Button_CreateParams_GetDefault_ReturnsExpected()
+        {
+            using var control = new SubButton();
+            CreateParams createParams = control.CreateParams;
+            Assert.Null(createParams.Caption);
+            Assert.Equal("Button", createParams.ClassName);
```

Comment:
I think in this and other similar tests we should do the following:
```suggestion
            Assert.Equal(Interop.ComCtl32.WindowClasses.WC_BUTTON, createParams.ClassName);
```

Reply:
I decided not to do this because I want to explicitly test the string and make sure out constants are correct. Same goes for the styles

Reply:
I don't think this is a correct approach. There are two different aspects here: 
1. ensure the literal value is correct, and 
2. the components reference the literal value.

If we want to test constants we should be testing `Interop.ComCtl32.WindowClasses` to ensure definitions are correct.
Testing an object other than object under test turns tests in to component/integration/etc tests, and thus immediately increase a maintenance overhead.

Suppose we follow the approach, and have a literal property which is referenced by a number of components. We write tests to ensure that all these components use the same literal (but not testing it via a reference). Then for whatever reason we need to change the value of the string, and we will have to update all these tests to reflect the change. This is not correct.

Reply:
But then how do we verify that the value of the string is correct?

Reply:
`Button` class references `Interop.ComCtl32.WindowClasses.WC_BUTTON`, and it takes the value of that const for granted:
![image](https://user-images.githubusercontent.com/4403806/68665037-75a16980-0595-11ea-9e81-74d2bae67511.png)
So in our tests we should not assert the value of `Interop.ComCtl32.WindowClasses.WC_BUTTON`, we should just assert that the classname is the same as the const.

Asserting the value of the const is completely orthogonal to this test.

Reply:
> But then how do we verify that the value of the string is correct?

If we really want to do this, we need to write tests for `Interop.ComCtl32.WindowClasses`, e.g. ``Interop.ComCtl32.WindowClassesTests` where we assert correctness of the literals.